### PR TITLE
[JBPM-8725] Invalid REST API return type - quated string instead of json/xml

### PR DESCRIPTION
**Pull request by elguardian**
changing the marshalling for returning proper json objects in case of primitive types

### CODE REVIEW

**Comment by elguardian** [comment]
@MarianMacik I don't see you as a reviewer.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java [hunk 1]
```diff
@@ -62,7 +76,6 @@ public static void buildAndDeployArtifacts() {
         KieServerDeployer.buildAndDeployCommonMavenParent();
         KieServerDeployer.buildAndDeployMavenProjectFromResource("/kjars-sources/rest-processes");
         // set the accepted formats with quality param to express preference
-        acceptHeadersByFormat.put(MarshallingFormat.JAXB, "application/xml;q=0.9,application/json;q=0.3");// xml is preferred over json
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java]: Why is this line removed?

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/IdentifierJbpmRestIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            logger.info("Value returned {} {}", serialized, getMediaType());
+            if (marshallingFormat.equals(MarshallingFormat.JAXB)) {
+                wrapId = marshaller.unmarshall(serialized, Long.class);
+                id = ((Long) wrapId);
+            } else {
+                try {
+                    wrapId = new JSONObject(serialized);
+                    id = ((JSONObject) wrapId).getString("value");
+                } catch (JSONException ex) {
+                    logger.error("expected json object from kie server", ex);
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/IdentifierJbpmRestIntegrationTest.java]: Please don't log errors in test, rather throw it making the test fail.

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/MarshallingFormat.java [hunk 3]
```diff
@@ -53,6 +57,19 @@ public static MarshallingFormat fromId(int id) {
         }
     }
 
+    public static boolean isStrictType(String type) {
+        int idx;
+        if((idx = type.indexOf(';')) < 0) {
+            return false;
+        }
+        // we map parameter=1, pararmeter=2,.... into a map
+        Map<String, String> parameters = Arrays.asList(type.substring(idx + 1).split(",")).stream().map(e -> e.split("=")).collect(Collectors.toMap(e -> ((String[]) e)[0], e -> ((String[]) e)[1]));
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/MarshallingFormat.java]: idx is the index of the occurrence of the character.... if I don't do idx + 1 it will add the character to the substring

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/MarshallingFormat.java [hunk 4]
```diff
@@ -53,6 +57,19 @@ public static MarshallingFormat fromId(int id) {
...
+        if (parameters.containsKey("strict")) {
+            return Boolean.parseBoolean(parameters.get("strict"));
+        }
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/MarshallingFormat.java]: done

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/IdentifierJbpmRestIntegrationTest.java (hunk 2 above)
Comment (elguardian) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/IdentifierJbpmRestIntegrationTest.java]: done

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java (hunk 1 above)
Comment (elguardian) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/JbpmRestIntegrationTest.java]: mistake

**Comment by elguardian** [comment]
Jenkins retest this please.

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/MarshallingFormat.java [hunk 5]
```diff
@@ -53,6 +57,20 @@ public static MarshallingFormat fromId(int id) {
...
 
+    public static boolean isStrictType(String type) {
+        int idx;
+        if ((idx = type.indexOf(';')) < 0 || (idx + 1) == type.length()) {
+            return false;
+        }
+        // we map parameter=1, pararmeter=2,.... into a map
+        Map<String, String> parameters = Arrays.asList(type.substring(idx + 1).split(","))
+                                               .stream()
+                                               .filter(e -> e.split("=").length > 1) // remove bad parameters
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/MarshallingFormat.java]: I would even say that it has to equal to 2. But that just a cosmetic point of view.

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/MarshallingFormat.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (MarianMacik) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/MarshallingFormat.java]: Thanks.

**Comment by sutaakar** [comment]
Seems that IdentifierJbpmRestIntegrationTest is failing for JSON

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/IdentifierJbpmRestIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        Map<String, Object> valuesMap = new HashMap<String, Object>();
+        valuesMap.put(CONTAINER_ID, resource.getContainerId());
+        valuesMap.put(PROCESS_ID, HUMAN_TASK_OWN_TYPE_ID);
+
+        Response response = null;
+        try {
+            WebTarget clientRequest = newRequest(build(TestConfig.getKieServerHttpUrl(), PROCESS_URI + "/" + START_PROCESS_POST_URI, valuesMap));
+            logger.info( "[POST] " + clientRequest.getUri());
+            // we set strict json parameter
+            MediaType type = getMediaType();
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/rest/IdentifierJbpmRestIntegrationTest.java]: Just noticed that this line is not needed here.

**Comment by elguardian** [comment]
fixed ... needed to change negotiation (backward compatible)

**Comment by elguardian** [comment]
@MarianMacik any progress ?

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by MarianMacik** [comment]
jenkins retest this
